package/metrics_daemon/src/metrics_daemon.c

```diff
@@ -0,0 +1,349 @@
+ */
+static int handle_walk_path(const char *fpath, const struct stat *sb, int tflag)
+{
+    if(first_folder == TRUE)     ///< skip the root since it is handled by main()
+    {
+        first_folder = FALSE;
+        return 0;
+    }
+    char * bname = extract_filename(fpath);
+    json_object * jobj_Cur = loop_through_folder_name(fpath,root_name, root_length, jobjRoot);  ///< get the json object needs to be updated
+    switch(tflag)
+    {
+        case FTW_D:
+        {
+            bool file_entry_exist = FALSE;
+            json_object_object_foreach(jobj_Cur, key, val) {
+                if(strcmp(bname,key)==0)
+                {
+                    file_entry_exist = TRUE;
+                    jobjCur = val;
+                    break;
+                }
+            }
+            if(!file_entry_exist)                 ///< if it is a folder and it doesn't exist in json tree, add a new folder, do nothing otherwise
+            {
+                json_object *jnewobj = json_object_new_object();     ///< Create a new object here to set the name of folder
+                json_object_object_add(jobj_Cur,bname, jnewobj);
+            }
+            break;
+        }
+
+        case FTW_F:
+        {
+            FILE * fp = fopen(fpath, "r");           ///< read file to get the data
+            char buf[64];
+            long file_len_long = sb->st_size-1;
+            int file_len = (int)(sb->st_size-1);
+            if(file_len<64 && file_len_long >=0)
+            {
+                fread(&buf[0],(unsigned int)file_len,1,fp);
+            } else{
+                file_len = 0;                        ///< if file is empty, set the str length to zero
+            }
+            fclose(fp);
+            bool file_entry_exist = FALSE;
+            json_object_object_foreach(jobj_Cur, key, val) {
+                if(strcmp(bname,key)==0)
+                {
+                    file_entry_exist = TRUE;
+                    json_object_set_string_len(val,&buf[0],file_len);     ///< if file exist, update
+                    break;
+                }
+            }
+            if(!file_entry_exist)     ///< add a new node if it doesn't exist
+            {
+                json_object_object_add(jobj_Cur,bname, json_object_new_string_len(&buf[0],file_len));
+            }
+            break;
+        }
+        default:
+        {
+            perror("FTW: unexpected file type\n");   ///< For now, only support file and forlder
+            return -1;
+        }
+    }
+
+    free(bname);  ///< free the file name allocated under loop_through_folder_name()
+    return 0;
+}
+/**
+ * @brief write_messaging_metrics_to_file
+ *
+ * Write messaging metrics to file
+ * @param metrics_dir: directory that holds the metrics directory tree structure
+ */
+static void write_messaging_metrics_to_file(const char *metrics_dir) {
+    (void) metrics_dir;
+    /* Logic to write metrics info here */
```

Comment:
Let's use a consistent style for one-line comments, either `// ...` or `/* ... */`